vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -95,6 +111,107 @@ def __preprocess(
 
         return request
 
+    def _make_tools_request(
+        self, request: ChatDetectionRequest
+    ) -> Union[ChatDetectionRequest, ErrorResponse]:
+        """Convert original chat detection request to Granite Guardian-compatible request with
+        tools, user, and assistant messages."""
+
+        if (
+            "risk_name" not in request.detector_params
```

Comment:
is this because otherwise, the "content" would become incompatible with other GG risks? 

What happens if we send the modified content to GG but the request looks like function_call ?